### ISSUE TITLE
Hologram Initial Visiblity

### ISSUE THREAD

**Issue by UltraFaceguy** [DecentSoftware-eu/DecentHolograms#11]
I'm moving from using holodisplays API to DH for my hologram stuff - something HD does better is that you can use the visibility manager stuff to change who can see it, before it's spawned in, so it doesn't appear then get hidden. With DH, if the user lags or there's a TPS spike for any reason, they'll see the hologram right as its created, before .hide() is reached.

Long story short, this is a request to allow created holograms to default to invisible, so they are never rendered, even for a moment, for players who haven't been shown the hologram with .show()

**Comment by d0by1** [comment]
Hello, this depends on the way you create the hologram. If you use DHAPI then yes, the hologram will be shown to everyone. But you can also create it with Hologram constructor and then use Hologram#enable() and Hologram#show(Player...) with the players that you want the hologram to be shown to.

**Comment by UltraFaceguy** [comment]
So I tried it out, and it turns out... This is false
`create it with Hologram constructor and then use Hologram#enable() and Hologram#show(Player...)`
The Holorgram#enable method actually does this, which is what tripped me up.
```
public void enable() {
    super.enable();
    this.showAll();
    this.register();
  }
```
It shows it to all by default. Additionally...
```
Hologram holo = new Hologram(UUID.randomUUID().toString(), location.clone(), false);
DHAPI.setHologramLines(holo, List.of(StringExtensionsKt.chatColorize(text)));
holo.setEnabled(true);
holo.register();
holo.show(player, 0);
```
The above should bypass the fact that enable() includes showAll(), however, in testing, other people can still see the holograms. The only way to show a short-term created hologram to only one person via the API seems to be my hacky page solution.

**Comment by d0by1** [comment]
Hi, I meant Hologram#setEnabled(true), my bad. Strange that it doesn't work as intended though. Anyway, I will look into it.

**Comment by UltraFaceguy** [comment]
Hrm... That's actually what I did for my first attempt, or close to it. It'll try that right now and report back.

**Comment by SkytAsul** [comment]
Any news regarding this? It's not this complicated to introduce a "defaultVisibility" parameter.

**Comment by UltraFaceguy** [comment]
Still have this issue :v

**Comment by SlimeDog** [comment]
We are starting a project to support DecentHolograms with pHD, which will solve this issue (and others). You can follow along at [https://github.com/SlimeDog/pHD/issues/4](https://github.com/SlimeDog/pHD/issues/4).

**Comment by d0by1** [comment]
Visibility is not handled very well in the current version of the plugin. It basically updates the visibility for all players automatically and the only way to avoid this is to set a view permission to the hologram. Anyway, this is all going to be solved in upcoming 3.0 rewrite of the plugin which is gonna provide a proper visibility management in the API. I will close this issue now as there is nothing else to discuss.
